pyo3-macros-backend/src/method.rs

```diff
@@ -20,13 +20,20 @@ use crate::{
 pub struct FnArg<'a> {
     pub name: &'a syn::Ident,
     pub ty: &'a syn::Type,
-    pub optional: Option<&'a syn::Type>,
-    pub default: Option<syn::Expr>,
-    pub py: bool,
     pub attrs: PyFunctionArgPyO3Attributes,
-    pub is_varargs: bool,
-    pub is_kwargs: bool,
-    pub is_cancel_handle: bool,
+    pub kind: FnArgKind<'a>,
+}
+
+#[derive(Clone, Debug)]
+pub enum FnArgKind<'a> {
+    Regular {
+        default: Option<syn::Expr>,
+        ty_opt: Option<&'a syn::Type>,
```

Comment:
Default is a soft keyword, and ty_opt is not a very descriptive name. Maybe something like this fits better:

```suggestion
        default_argument: Option<syn::Expr>,
        option_wrapped_type: Option<&'a syn::Type>,
```

Or whatever you think is most descriptive